src/main/java/com/conveyal/datatools/manager/jobs/feedmerge/FeedMergeContext.java

```diff
@@ -0,0 +1,158 @@
+package com.conveyal.datatools.manager.jobs.feedmerge;
+
+import com.conveyal.datatools.manager.DataManager;
+import com.conveyal.datatools.manager.auth.Auth0UserProfile;
+import com.conveyal.datatools.manager.models.FeedVersion;
+import com.conveyal.datatools.manager.utils.MergeFeedUtils;
+import com.conveyal.gtfs.loader.Feed;
+import com.conveyal.gtfs.loader.Table;
+import com.google.common.collect.Sets;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Contains merge information between an active feed and a future feed.
+ */
+public class FeedMergeContext implements Closeable {
+    public final List<FeedToMerge> feedsToMerge;
+    public final FeedToMerge activeFeedToMerge;
+    public final FeedToMerge futureFeedToMerge;
+    public final Set<String> activeTripIds;
+    public final Set<String> futureTripIds;
+    public final boolean serviceIdsMatch;
+    public final boolean tripIdsMatch;
+    public final Feed futureFeed;
+    public final Feed activeFeed;
+    public final LocalDate activeFeedFirstDate;
+    private LocalDate futureFeedFirstDate;
+    private LocalDate futureFirstCalendarStartDate = LocalDate.MAX;
+    /**
+     * Trip ids shared between the active and future feed.
+     */
+    public final Set<String> sharedTripIds;
+    /**
+     * Holds the auto-generated agency id to be updated if none was provided.
+     */
+    private String newAgencyId;
```

Comment:
@br648 The bug for this was introduced in a different PR (basically, agencyId would never get updated where needed) and I fixed it here, would you recommend adding a test for it?

Reply:
As discussed a test would be useful. It would also be useful to see how two feeds are merged that have the same agency but no agency id e.g. does this create two new rows in the merged table for the same agency with differenent ids?

Reply:
Finally handled it in https://github.com/ibi-group/datatools-server/pull/376/commits/39db651a614de38be62f1b5c3fd2b23cb6fb70c8 with a new test and supporting data!